.github/workflows/pythonpackage.yml

```diff
@@ -61,3 +61,18 @@ jobs:
       run: python3 -m pip install -r doc/requirements.txt -e .
     - name: Build docs
       run: make -C doc html
+
+  mypy:
+    continue-on-error: true
```

Comment:
It looks a little strange to see "Python package / mypy (pull_request)" as the title for this check.  I'm leaning towards moving it to a separate workflow file.  Or is there any particular reason why it should be linked with the `pythonpackage.yml`?

Either way, we should give it a more appealing name such as `- name: Run mypy static type checker`

For the possibly new workflow file, we could use a more generic name / title and keep in mind that more checkers / linters might be added there later.  Something like "Check correctness" or "Static code analysis".